src/main/java/com/minecolonies/coremod/research/UnlockAbilityResearchEffect.java

```diff
@@ -13,36 +14,36 @@
     /**
      * Whether the effect has been unlocked or not.
      */
-    private boolean unlocked;
+    private int level;
```

Comment:
before we had the two different "effects" which were unlock yes/No or unlock with value. Now we can combine them as int as you kinda did here which makes me wonder if we actually need different "effects". As the way I see it its just a rich information return of the research, and not the actual effect of the research. In that case we wouldnt need effect lists at all/neither any of the current "effect" implementation

Reply:
There's technically four, split between the Addition and Multiplication floats, and the UnlockAbility, UnlockBuilding booleans.  From a logical perspective they can definitely be merged into a single implementation; it's as much a matter from presentation and programming maintenance that it's a question.  Addition and Multiplication have different approaches to translation arguments due to different magnitudes of their most common values, for example.

Definitely can be reduced down if that simplifies matters for you all, though.